### ISSUE TITLE
help not displayed if required field has not been set

### ISSUE THREAD

**Issue by andrewbhamilton** [FlorianRappl/CmdParser#7]
Hi there

Firstly, thank you for this very helpful library.

I found that help was not being outputted if a required argument was not specified on the cmd line. Is this the desired behaviour?

I personally did not want this behaviour so have changed it so that help is always displayed (if -h/--help is specified) but a required argument is not.

I would be happy to create a PR if you wish to view my changes. I just swapped the order in which you check for unhandled required arguments and parse the handled ones in "run". This may screw it up elsewhere as I have not run the tests, but it works for me!

Cheers

**Comment by FlorianRappl** [comment]
Yes, this is not the desired behavior. I thought I fixed it in #4, but as you can read in that issue the bug still exists. I did not have time for the changes mentioned in the conversation from #4.

A PR would be much welcome! Thanks already!

**Comment by RobertBakaric** [comment]
you can simply do what i suggested in [#6](https://github.com/FlorianRappl/CmdParser/issues/6) evaluate if -h/--help has been called and if it was, then do not evaluate the required fields required fields:
<code>

```
bool run(std::ostream& output, std::ostream& error) {    
bool help =false;  // add this
if (_arguments.size() > 0) {
    auto current = find_default();

/*  Add this line*/
    if ( std::find(_arguments.begin(), _arguments.end(), "--help") != _arguments.end() ||
        std::find(_arguments.begin(), _arguments.end(), "-h") != _arguments.end())

            help = true; 
```

...

```
        for (auto command : _commands) {
            if (command->required && !command->handled && help == false) { // here 
                error << howto_required(command);
                return false;
            }
        } 
```

...
</code>

but this is just a quick "fix" for help only. in this case every time you press -h od --help it prints usage no matter to what other options are set or not set (without  messing with the original code too much)

**Comment by FlorianRappl** [comment]
This was finally fixed with PR #8.
